pandora-calculation/src/test/java/com/science/earth/biogeochemistry/freshwaters/pandora/general/services/implementations/CellTimestepServiceImplIntegrationTest.java

```diff
@@ -71,4 +78,10 @@ void testCalculateNextTimestep() {
 	cellTimestepService.calculateNextTimestep(CELL, T_0);
 	Assertions.assertArrayEquals(Y_END,  yMapService.findAtCellAndTimestep(CELL, T_0.plusYears(1)), 1e-3);
     }
+
+    @Test
+    void testCalculateTimeSeries() {
```

Comment:
Here I actually would like to see how the method performs in a series, with a number of timesteps more than one (argument 3). So although it totally covers all the lines in code, it doesn't cover full functionality. How does it respond when you put in 0, or for either Cell or LocalDateTIme a null?

It fails in a very ugly way now, it should throw a clean exception instead of a NullPointerException.
To the letter of the task you fullfilled it, but it made me realize that the calculateTimeSeries() is far from finished. For a future ticket Ill come up with more specific scenarios.

For now I think it is just fine, but it made me realize this method needs a more lenient functionality.

Reply:
Hi @WJ-van-Hoek  maybe we can add more  validation steps for null checks and for objects field we can definitely use optional() type of object declaration
That way we can triage any kind of nullpointerexception for null objects but adding extra validation check at the validate()  class at controller layer will help

Reply:
Yes, I agree about the validation and optionals.

Validation is something i haven't touched upon at all in the code base. I will have to think about it how to prioritize that at this moment, before I want to make specific decisions though. Hopefully I can make a bit more time this weekend to think it out. I also need some time though for a refactor in another repository of mine; once that is finished I will come up with some ideas.

For now, if you want to work on other test tickets, go ahead as much as you like.